sktime/transformations/series/holiday/_holidayfeats.py

```diff
@@ -306,35 +319,39 @@ def _generate_holidays(
             if name in holidays_by_name:
                 dates = holidays_by_name[name]
             else:
-                raise ValueError(f"holiday: {name} not found in calendar.")
-
+                if raise_warnings and warning_instance is not None:
+                    warnings.warn(
+                        message=(
+                            f"{warning_instance.__class__.__name__}: "
+                            f"Holiday '{name}' not found in calendar. Skipping."
+                        ),
+                        stacklevel=2,
+                    )
+
+            # We then get the number of days before and after
+            # the holiday.
+            before, after = window
+            neg_before = -before
             # For each holiday, we iterate over dates.
             for dte in dates:
-                # We then get the number of days before and after
-                # the holiday.
-                before, after = window
-
                 # Finally, we add all days within the window to the
                 # holiday, making sure that we do not overwrite
                 # already existing holidays.
-                msg = (
-                    "Holiday already exists. Please make sure holiday windows "
-                    "are not overlapping with other holidays."
-                )
-
-                for days in range(1, before + 1):
-                    date_before = dte - datetime.timedelta(days=days)
-                    if date_before not in holidays_by_date:
-                        holidays_by_date[date_before] = name
-                    else:
-                        raise ValueError(msg)
-
-                for days in range(1, after + 1):
-                    date_after = dte + datetime.timedelta(days=days)
-                    if date_after not in holidays_by_date:
-                        holidays_by_date[date_after] = name
+                for days in range(neg_before, after + 1):
+                    date_window = dte + datetime.timedelta(days=days)
+                    if date_window not in holidays_by_date:
```

Comment:
could you explain what this block does? Not sure I get the "conflict" warning condition.

Reply:
This line checks if the selected day from the window is present in the current holidays dictionary, if it is not present already we add it to the dictionary and this helps to avoid overriding the current existing days.